pkg/controllerutils/mapper/enqueue_mapped.go

```diff
@@ -56,11 +53,13 @@ func (f MapFunc) Map(ctx context.Context, log logr.Logger, reader client.Reader,
 // behavior on UpdateEvents.
 // For UpdateEvents, the given UpdateBehavior decides if only the old, only the new or both objects should be mapped
 // and enqueued.
-func EnqueueRequestsFrom(m Mapper, updateBehavior UpdateBehavior, log logr.Logger) handler.EventHandler {
+func EnqueueRequestsFrom(ctx context.Context, mgr manager.Manager, m Mapper, updateBehavior UpdateBehavior, log logr.Logger) handler.EventHandler {
 	return &enqueueRequestsFromMapFunc{
 		mapper:         m,
 		updateBehavior: updateBehavior,
+		ctx:            ctx,
 		log:            log,
+		reader:         mgr.GetCache(),
```

Comment:
Isn't it better to pass the cache instead of the manager? Will this work for cases where the watch is for objects in a different cluster and hence not the manager cache?

Reply:
Make sense. Adapted the `EnqueueRequestsFrom` function.

Reply:
@timebertt Do you know how this works for the above case? For example the watch for backupbucket objects in the backupentry controller, where the manager cache is actually the seec cluster cache, but the backupbucket objects are in the garden cluster. Is it enough that we pass the gardencluster cache as the Source? https://github.com/gardener/gardener/blob/cf2e024732cfb42e341be78f261c5102b9a7edbc/pkg/gardenlet/controller/backupentry/add.go#L87

Reply:
Yes, that should be enough. There's nothing else to inject, I guess.